driver-async/src/main/com/mongodb/async/client/MongoIterableSubscription.java

```diff
@@ -67,7 +68,17 @@ boolean checkCompleted() {
     void postTerminate() {
         try {
             if (batchCursor != null) {
-                batchCursor.close();
+                boolean closeCursor = false;
+                synchronized (this) {
+                    if (!isReading) {
```

Comment:
Nit: conditionals are generally easier to read if negations are minimized.  So consider reversing the condition to just `isReading`.

Reply:
I have reversed the condition.